### PR TITLE
net: lib: nrf_cloud: cleanup header

### PR DESCRIPTION
**Pull request by jayteemo**
Cleanup header comments so that proper documentation is produced.
Remove unused items.
IRIS-5092

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

**Comment by jayteemo** [comment]
this is mostly for `nrf_cloud.h` and `nrf_cloud_rest.h` but i updated the others to be consistent.

**Review by glarsennordic — APPROVED** [review]

File: include/net/nrf_cloud.h [hunk 1]
```diff
@@ -175,17 +195,18 @@ enum nrf_cloud_connect_result {
...
-/**@ nRF Cloud error codes.
+/** @brief nRF Cloud error codes.
  * See the <a href="https://api.nrfcloud.com/v1#section/Error-Codes">Error Codes</a>
  * section of nRF Cloud API documentation for more information.
  */
 enum nrf_cloud_error {
 	NRF_CLOUD_ERROR_UNKNOWN			= -1,
 	NRF_CLOUD_ERROR_NONE			= 0,
-	/** nRF Cloud API error codes */
+	/* nRF Cloud API defined error codes below */
```
Comment (peknis) [inline include/net/nrf_cloud.h:209]: I guess this is not intended to be shown?

File: include/net/nrf_cloud.h [hunk 2]
```diff
@@ -198,7 +219,7 @@ enum nrf_cloud_error {
 	NRF_CLOUD_ERROR_NO_DEV_NOT_PROV		= 40412,
 	NRF_CLOUD_ERROR_NO_DEV_DISSOCIATE	= 40413,
 	NRF_CLOUD_ERROR_NO_DEV_DELETE		= 40414,
-	/** Item was not found. No error occured, the requested item simply does not exist */
+	/* Item was not found. No error occurred, the requested item simply does not exist */
```
Comment (peknis) [inline include/net/nrf_cloud.h:222]: I guess this is not intended to be shown?

**Review by peknis — APPROVED** [review]

File: include/net/nrf_cloud.h (hunk 2 above)
Comment (jayteemo) [reply include/net/nrf_cloud.h:222]: Correct, these are documented in the cloud docs.  Just making a note here in the code.

**Review by rlubos — APPROVED** [review]

**Pull request merged**